### DISCUSSION TITLE
Date query not returning expected results

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#14840] [General]
Hi,

I have a JavaScript index which indexes a date field stored as an ISO date / time string, the value stored for the document in question is

`"moveAtUTC": "2022-09-30T10:00:00.000Z"`

I can see this value in the terms for the index field and can query without a date filter and return the document, however when I issue this query, nothing is returned

`from index 'Movements' where organisationId = "organisations/131-A" and moveAtUTC < "2022-09-30T23:59:59.999Z"`

If I change the query and add 1 day I get the doc returned in the results

`from index 'Movements' where organisationId = "organisations/131-A" and moveAtUTC < "2022-09-31T23:59:59.999Z"`

What am I missing here, is there any conversion of the date field going on which could cause this query to fail?

Thanks

**Comment by garayx** [comment]
Hi 

Can you please submit a sample app that demonstrate the issue, or a failing unit test: https://ravendb.net/docs/article-page/5.4/csharp/start/test-driver#getting-started-writing-your-unit-test-using-testdriver

**Comment by ayende** [comment]
Something to note, you have `...00.000Z"` - what happens if you use 7 digits after the `.` ? That is the .NET format.

**Reply by nickchampion** [reply]
Hi @ayende yes that works with the 7 digit precision, thanks.

Seems like this is something the database should handle though, in particular for those of us not using the .NET client?

**Reply by ayende** [reply]
One of the supported data time formats we have is: `'yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fff'Z'`, so that should be fine. 
I wonder if this is related to work done at query time, but not sure. 
Can you send us a repro test for this? I tried, but it works for me:

![image](https://user-images.githubusercontent.com/116915/188098878-26d93cb5-d128-4da5-8d1c-f1403ce520f0.png)
